mypy/newsemanal/semanal_main.py

```diff
@@ -168,10 +168,12 @@ def process_top_levels(graph: 'Graph', scc: List[str], patches: Patches) -> None
         worklist += list(reversed(core_modules)) * CORE_WARMUP
     final_iteration = False
     iteration = 0
+    analyzer = state.manager.new_semantic_analyzer
+    analyzer.deferral_debug_context.clear()
```

Comment:
This doesn't work correctly inside import cycles, since we only clear one of the debug contexts.

Reply:
I was mistaken. There is only one semantic analyzer instance.